File: _CONTAINER/_BY_MONTH/_Feb/Thesis-physics-master/doc/particles/particle.js
/**
 * @fileoverview Particle object.
 * @author sebastian.poreba@gmail.com (Sebastian Poreba)
 */

goog.provide('smash.Particle');



/**
 * @struct
 * @constructor
 */
smash.Particle = function() {
  /**
   * @type {number}
   */
  this.positionX = 0.1;

  /**
   * @type {number}
   */
  this.positionY = 0.1;

  /**
   * @type {number}
   */
  this.velocityX = 0.1;

  /**
   * @type {number}
   */
  this.velocityY = 0.1;

  /**
   * @type {number}
   */
  this.age = 0;

  /**
   * In seconds.
   * @type {number}
   */
  this.lifespan = 0;

  /**
   * @type {boolean}
   */
  this.isDead = false;
};


/**
 *
 */
smash.Particle.prototype.step = function() {
  this.positionX += this.velocityX;
  this.positionY += this.velocityY;
  this.age++;
};


/**
 * Recover defaults.
 */
smash.Particle.prototype.reset = function() {
  this.positionX = 0;
  this.positionY = 0;
  this.velocityX = 0;
  this.velocityY = 0;
  this.age = 0;
  this.lifespan = 0;
  this.isDead = false;
};

File: _BY_MONTH/_Feb/Thesis-physics-master/doc/particles/particle.cpp
/**
 * @fileoverview Particle object.
 * @author sebastian.poreba@gmail.com (Sebastian Poreba)
 */
#include "particle.h"

smash::Particle::Particle() {
  positionX = 0;
  positionY = 0;
  velocityX = 0;
  velocityY = 0;
  age = 0;
  lifespan = 0;
  isDead = false;
}

/**
 * @param deltaTime
 */
void smash::Particle::step() {
  this->positionX += this->velocityX;
  this->positionY += this->velocityY;
  this->age++;
};


/**
 * Recover defaults.
 */
void smash::Particle::reset() {
  this->positionX = 0;
  this->positionY = 0;
  this->velocityX = 0;
  this->velocityY = 0;
  this->age = 0;
  this->lifespan = 0;
  this->isDead = false;
};
